From 8c141e53d720b14a1d8595e0f756f1619df6e763 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Wed, 16 Apr 2025 11:42:54 +0200
Subject: [PATCH] tools: m4: set -std=gnu17

Fedora 42 updated to GCC15 which now defaults to GNU23 as the default
instead of GNU17[1], and this breaks m4 compilation.

Its been reported upstream [2], so until its fixed lets simply set C
language version back to GNU17.

[1] https://gcc.gnu.org/gcc-15/porting_to.html#c23
[2] https://savannah.gnu.org/support/?111150

Link: https://github.com/openwrt/openwrt/pull/18505
Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 tools/m4/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/m4/Makefile b/tools/m4/Makefile
index ee369f3cbb3d99..c13155397d25f3 100644
--- a/tools/m4/Makefile
+++ b/tools/m4/Makefile
@@ -18,6 +18,7 @@ HOST_BUILD_PARALLEL:=1
 
 include $(INCLUDE_DIR)/host-build.mk
 
+HOST_CFLAGS += -std=gnu17
 HOST_CONFIGURE_VARS += gl_cv_func_strstr_linear=no
 
 define Host/Uninstall
